### ISSUE TITLE
CDATA hang

### ISSUE THREAD

**Issue by weasel009** [lexborisov/myhtml#156]
Hi Alexander,

It appears that the tokenizer hangs if **<![CDATA[** shows up in HTML.
 tokenizer.c:
```
    // CDATA sections can only be used in foreign content (MathML or SVG)
    if(strncmp(tagname, "[CDATA[", 7) == 0) {
        if(tree->current_qnode->prev && tree->current_qnode->prev->args)
        {
            myhtml_tree_wait_for_last_done_token(tree, tree->current_qnode->prev->args);
            myhtml_tree_node_t *adjusted_current_node = myhtml_tree_adjusted_current_node(tree);
```

I attach a sample file to reproduce the problem.
[fourmilab-ch-babbage-pascal.txt](https://github.com/lexborisov/myhtml/files/2353589/fourmilab-ch-babbage-pascal.txt)

Regards!

**Comment by lexborisov** [comment]
Hello @weasel009 !
I understood what the problem is, tomorrow I'll try to fix it.
Thanks!

**Comment by lexborisov** [comment]
@weasel009 
Fixed in [c97bfba](https://github.com/lexborisov/myhtml/commit/c97bfba6e7d00747e543a88791f647ac0e500e00) commit.
